homeassistant/components/screenlogic/water_heater.py

```diff
@@ -0,0 +1,127 @@
+"""Support for a ScreenLogic Water Heater."""
+import logging
+
+from screenlogicpy.const import HEAT_MODE
+
+from homeassistant.components.water_heater import (
+    SUPPORT_OPERATION_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+    WaterHeaterEntity,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS, TEMP_FAHRENHEIT
+
+from . import ScreenlogicEntity
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORTED_FEATURES = SUPPORT_TARGET_TEMPERATURE | SUPPORT_OPERATION_MODE
+
+HEAT_MODE_NAMES = HEAT_MODE._names  # pylint: disable=protected-access
```

Comment:
Can you make names public in the library to avoid the pylint exception?

Reply:
Yea, though likely I'll want to change it library-side again in the future. Wanting something more elegant there.

Reply:
Should be resolved with commit [9bece4b](https://github.com/home-assistant/core/pull/47933/commits/9bece4b9bb93ae8f30ac4fe6c53f8c10d105f127)